--- a/plat/mediatek/mt7986/drivers/pll/pll.c
+++ b/plat/mediatek/mt7986/drivers/pll/pll.c
@@ -82,13 +82,13 @@ void mtk_pll_init(int skip_dcm_setting)
 	/* Set PLL frequency */
	if ((mmio_read_32(IAP_REBB_SWITCH) & IAP_IND) == 0) {
		mmio_write_32(ARMPLL_CON1, 0x50000000); /* 1.6G */
	}
	else {
		if ((mmio_read_32(BS_PAD_SYS_WATCHDOG) & BS_CPU_VCORE))
-			mmio_write_32(ARMPLL_CON1, 0x64000000); /* 1.023: 2G */
+			mmio_write_32(ARMPLL_CON1, 0x78000000); /* 1.20: 2.4G */
		else
			mmio_write_32(ARMPLL_CON1, 0x50000000); /* 0.85: 1.6G */
	}

	mmio_setbits_32(ARMPLL_CON0, 0x104); /* 1.6G or 2G */

